app/main.py

```diff
@@ -1,7 +1,37 @@
-def get_weather() -> None:
-    # write your code here
-    pass
+import os
+import requests
+from dotenv import load_dotenv
+
+
+def get_weather(key: str, city: str = "Paris") -> None:
+    url = "https://api.weatherapi.com/v1/current.json"
+    response = requests.get(
+        url, params={
+            "key": key,
+            "q": city
+        }
+    )
+    if response.status_code == 200:
+        weather_data = response.json()
+        location_localtime = weather_data["location"]["localtime"]
+        location_name = weather_data["location"]["name"]
+        location_country = weather_data["location"]["country"]
+        location_weather = weather_data["current"]["condition"]["text"]
+        location_wind = weather_data["current"]["wind_mph"]
+        temp_c = weather_data["current"]["temp_c"]
+
+        print(
+            f"[{location_localtime}] {location_name} / {location_country}\n"
+            f"Weather: {temp_c} C, {location_weather}, {location_wind} mph\n"
+        )
+    else:
+        print(f"Failed to get weather data: {response.status_code}")
```

Comment:
Why Use a try Block Here?
Network Errors: The requests.get call can fail due to network issues, timeouts, or invalid URLs. Wrapping it in a try block ensures that these errors are caught and handled gracefully rather than causing your program to crash.

HTTP Errors: If the API returns an error response (e.g., 4xx or 5xx status codes), using response.raise_for_status() will raise an exception. This exception is caught by the except block to provide a meaningful error message.

JSON Parsing Errors: The response.json() call can fail if the response is not valid JSON. Wrapping this in a try block allows you to catch ValueError and handle it appropriately.

Missing Keys: The code assumes that certain keys will always be present in the JSON response. If the API response format changes or if the data is missing, this can raise a KeyError, which you can catch and handle to avoid crashes.

Unforeseen Errors: Catching a generic Exception ensures that any other unexpected errors are also handled, providing robustness to your function.

Reply:
Got it